r2-core/src/test/java/com/linkedin/r2/transport/http/client/ratelimiter/TestConstantQpsRateLimiter.java

```diff
@@ -88,21 +92,57 @@ public void eventLoopStopsWhenTtlExpiresAllRequests()
     Assert.assertSame(executor.getExecutedTaskCount(), prevTaskCount);
   }
 
+  @Test
+  public void ensureRandomButConstantRate()
+  {
+    ClockedExecutor executor = new ClockedExecutor();
+    ConstantQpsRateLimiter rateLimiter =
+        new ConstantQpsRateLimiter(executor, executor, executor, TestEvictingCircularBuffer.getBuffer(executor));
+    rateLimiter.setRate(200d, ONE_SECOND, 1);
```

Comment:
If i change this to 0.05d per second, and have executor run for 200s, the test was stopped after just 1 execution. I think we should use a different executor for circular buffer as in your previous test cases: `TestEvictingCircularBuffer.getBuffer(circularBufferExecutor));`

Reply:
Done